### PR TITLE
Use __array_wrap__ instead of __array_prepare__

### PR DESCRIPTION
**Pull request by mhvk**
### Description

This pull request is to address the removal of `__array_prepare__` in numpy 2.0 (see https://github.com/numpy/numpy/pull/25105, where it was noticed that astropy uses it). Generally, `__array_wrap__` should be used; I think the use of `__array_prepare__` was just an oversight. Note that if I just remove `__array_prepare__`, the only thing that fails is
```
docs/nddata/ccddata.rst F                                                                                                             [100%]

================================================================= FAILURES ==================================================================
___________________________________________________________ [doctest] ccddata.rst ___________________________________________________________
076 
077 Getting Data Out
078 ++++++++++++++++
079 
080 A `~astropy.nddata.CCDData` object behaves like a ``numpy`` array (masked if the
081 `~astropy.nddata.CCDData` mask is set) in expressions, and the underlying
082 data (ignoring any mask) is accessed through the ``data`` attribute:
083 
084     >>> ccd_masked = CCDData([1, 2, 3], unit="adu", mask=[0, 0, 1])
085     >>> 2 * np.ones(3) * ccd_masked   # one return value will be masked
Expected:
    masked_array(data=[2.0, 4.0, --],
                 mask=[False, False,  True],
           fill_value=1e+20)
Got:
    array([2., 4., 6.])
```

It would probably be good to have more tests, but since `nddata` is not something I usually deal with, it would be better if one of the nddata maintainers were to add this - please feel free to push to this branch (or take it over).

p.s.  I noticed that the `__array__` definition just ahead of `__array_prepare__` does not include the usual `dtype` argument -- see https://numpy.org/doc/stable/reference/generated/numpy.ndarray.__array__.html -- probably nice to address this at some point.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
About the "p.s." -- Should a new issue be opened?

**Review by pllim — APPROVED** [review]
A one-liner should not be controversial, right, right? I am approving so this can go in RC2. If Matt or Brett disagrees, they can open up follow-up PRs. Thanks!

**Comment by mhvk** [comment]
Thanks, @pllim. I think this is indeed uncontroversial and 6.0.rc2 is good in the sense that we might as well try to be as numpy 2.0 compatible as possible. As for better test cases, I leave that to the nddata maintainers...

**Pull request merged**
